### PR TITLE
feat(#913): add local runtime and production checks to vibew doctor

### LINKED ISSUE #913: feat: vibew doctor should diagnose runtime state, not just config syntax
## From v0.11.0 retrospective

\`vibew doctor\` validates config but doesn't check runtime state. After a deploy, it should tell you:

- \"WAF is configured but not active in the route chain\"
- \"TLS cert issued but proxy.started log reports tls_enabled: false\"
- \"Health check endpoint unreachable on the configured port\"

Walking the ejected config and comparing to the declared config is mechanical. One line of output would save an hour of debugging.

### PR DESCRIPTION
**Pull request by tibtof**
Closes #913

## Summary

- Added two new check layers to `vibew doctor` beyond config/Docker:
  - **Local Runtime**: upstream reachability via HTTP health check, TLS certificate validity (expiry check for self-signed certs)
  - **Production** (requires `--target` flag): SSH connectivity, remote container health via `docker compose ps`, domain DNS resolution against target IP, remote TLS certificate expiry
- Added `--target` and `--ssh-key` flags to `vibew doctor` CLI command
- Added `WithRemoteExecutor()` method to `DoctorService` for optional SSH executor injection
- Output is grouped with section headers ("Config & Docker", "Local Runtime", "Production") in both human and JSON output
- Production checks only run when `--target` is provided; graceful handling when SSH fails

## Test plan

- [x] All 11 original doctor tests pass unchanged
- [x] 15 new tests added covering:
  - Upstream reachable / unreachable
  - TLS cert valid / expired / expiring soon / missing / non-self-signed skip
  - SSH connectivity success / failure
  - Remote container health (healthy / unhealthy)
  - Auto-detection: no target = no remote checks
  - Section headers in human and JSON output
- [x] `make check` passes (golangci-lint 0 issues, all tests with race detector)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/app/ops/doctor.go [hunk 1]
```diff
@@ -275,17 +325,345 @@ func (s *DoctorService) checkContainerHealth(ctx context.Context, composePath st
...
+		Name:     "Remote containers",
+		Severity: SeverityOK,
+		Detail:   fmt.Sprintf("%d container(s) running", total),
+	}
+}
+
+// checkDomainDNS resolves the configured TLS domain and verifies it points to
+// the target host IP. Uses net.LookupHost from stdlib.
+func checkDomainDNS(domain, target string) CheckResult {
+	addrs, err := net.LookupHost(domain)
```
Comment (tibtof) [inline internal/app/ops/doctor.go]: **Missing context.Context**: `checkDomainDNS` performs DNS lookups via `net.LookupHost` (I/O) but does not accept a `context.Context` parameter. Per project rules, `context.Context` must be the first argument on all I/O functions. This also means there is no way for the caller to cancel the DNS lookup if it hangs.

**Suggestion**: Change the signature to `checkDomainDNS(ctx context.Context, domain, target string) CheckResult` and use `net.Resolver{}.LookupHost(ctx, ...)` instead of the package-level `net.LookupHost`.

File: internal/app/ops/doctor.go [hunk 2]
```diff
@@ -275,17 +325,345 @@ func (s *DoctorService) checkContainerHealth(ctx context.Context, composePath st
...
+				return host
+			}
+			return after
+		}
+	}
+	return raw
+}
+
+// checkRemoteTLSCert connects to domain:443 and checks the certificate expiry.
+func checkRemoteTLSCert(domain string) CheckResult {
```
Comment (tibtof) [inline internal/app/ops/doctor.go]: **Missing context.Context**: `checkRemoteTLSCert` performs a TLS dial (I/O) but does not accept a `context.Context` parameter. Per project rules, `context.Context` must be the first argument on all I/O functions. The 10-second timeout is hardcoded and cannot be cancelled by the parent context.

**Suggestion**: Change the signature to `checkRemoteTLSCert(ctx context.Context, domain string) CheckResult` and use a `tls.Dialer` with `DialContext` (Go 1.15+) or use `net.Dialer{Timeout: ..., Deadline: ctx.Deadline()}` to respect the context.

File: internal/app/ops/doctor.go [hunk 3]
```diff
@@ -275,17 +325,345 @@ func (s *DoctorService) checkContainerHealth(ctx context.Context, composePath st
...
+	return CheckResult{
+		Name:     "Remote containers",
+		Severity: SeverityOK,
+		Detail:   fmt.Sprintf("%d container(s) running", total),
+	}
+}
+
+// checkDomainDNS resolves the configured TLS domain and verifies it points to
+// the target host IP. Uses net.LookupHost from stdlib.
+func checkDomainDNS(domain, target string) CheckResult {
```
Comment (tibtof) [inline internal/app/ops/doctor.go]: **Missing unit tests**: `checkDomainDNS` and `checkRemoteTLSCert` are the only new check functions without any unit tests. These functions make real network calls (DNS lookup, TLS dial to port 443), which makes them untestable in the current form.

Because they bypass the port/adapter pattern and call `net.LookupHost` and `tls.DialWithDialer` directly, consider one of:
1. Extract the DNS resolution and TLS dialing behind small interfaces (or function parameters) that can be faked in tests.
2. At minimum, add a unit test for `extractHostFromTarget` (which is pure logic and easy to test with a table-driven test covering `ssh://user@host`, `ssh://user@host:22`, bare host, etc.).

File: internal/app/ops/doctor.go [hunk 4]
```diff
@@ -275,17 +325,345 @@ func (s *DoctorService) checkContainerHealth(ctx context.Context, composePath st
...
+	return CheckResult{
+		Name:     "Domain DNS",
+		Severity: SeverityWarn,
+		Detail:   fmt.Sprintf("%s resolves to %s but target is %s", domain, strings.Join(addrs, ", "), targetIP),
+	}
+}
+
+// extractHostFromTarget parses the host component from an ssh://user@host[:port]
+// URL. If parsing fails it returns the raw input.
+func extractHostFromTarget(raw string) string {
```
Comment (tibtof) [inline internal/app/ops/doctor.go:578]: **Prefer `net/url.Parse` over manual string splitting**: `extractHostFromTarget` manually splits on `://`, then on `@`, then uses `net.SplitHostPort`. The stdlib `net/url.Parse` handles all of these cases correctly and is more robust against edge cases (e.g., IPv6 addresses in brackets).

**Suggestion**:
```go
func extractHostFromTarget(raw string) string {
    u, err := url.Parse(raw)
    if err != nil || u.Host == "" {
        return raw
    }
    host := u.Hostname() // strips port and brackets for IPv6
    if host == "" {
        return raw
    }
    return host
}
```

This also needs a unit test -- at minimum a table-driven test covering: `ssh://user@host`, `ssh://user@host:22`, `ssh://host`, bare hostname, and IPv6 (`ssh://user@[::1]:22`).

File: internal/app/ops/doctor_test.go [hunk 5]
```diff
@@ -27,6 +36,47 @@ func (f *fakePortChecker) IsPortAvailable(_ context.Context, _ string, port int)
...
+func (f *fakeRemoteExecutor) Transfer(_ context.Context, _, _ string, _ bool) error {
+	return nil
+}
+
+func (f *fakeRemoteExecutor) TransferFile(_ context.Context, _, _ string) error {
+	return nil
+}
+
+func (f *fakeRemoteExecutor) DryRunTransfer(_ context.Context, _, _ string) ([]string, error) {
+	return nil, nil
```
Comment (tibtof) [inline internal/app/ops/doctor_test.go]: **Nil slice return**: Per project convention, slices should never be returned as nil when empty -- use `[]string{}` instead. This is a test fake, so the impact is low, but it sets a bad example.

**Suggestion**: `return []string{}, nil`

**Review by tibtof — COMMENTED** [review]
## Review Summary -- CHANGES REQUESTED

Overall this is a well-structured PR that adds valuable diagnostic checks to `vibew doctor`. The three-layer approach (Config & Docker, Local Runtime, Production) is clean, the existing tests all pass (30/30), and the new tests are thorough for the checks that go through ports.

### Issues found (4 actionable, 1 minor)

**1. Missing `context.Context` on I/O functions (code quality violation)**
Both `checkDomainDNS` and `checkRemoteTLSCert` perform network I/O (DNS lookups, TLS dials) but do not accept `context.Context` as their first parameter. Per project rules and Go idioms, all I/O functions must accept context for cancellation. The hardcoded timeouts cannot be overridden by the parent context.

**2. Missing unit tests for `checkDomainDNS` and `checkRemoteTLSCert`**
These are the only new check functions without any test coverage. Because they make real network calls directly (bypassing ports), they cannot be tested with fakes. Either inject the DNS resolver and TLS dialer as dependencies, or at minimum extract and test the pure logic (`extractHostFromTarget`).

**3. `extractHostFromTarget` has no unit test and uses manual URL parsing**
This function should use `net/url.Parse` from stdlib (handles IPv6, edge cases). It also needs a table-driven test covering `ssh://user@host`, `ssh://user@host:22`, bare hostname, and IPv6 forms.

**4. Minor: `fakeRemoteExecutor.DryRunTransfer` returns nil slice**
Per project convention: `return []string{}, nil` instead of `return nil, nil`.

### What looks good

- Correct reuse of existing `HealthChecker` and `RemoteExecutor` ports -- no unnecessary new interfaces
- `WithRemoteExecutor` builder pattern is clean and avoids changing the constructor signature
- Graceful error handling throughout -- SSH failures produce FAIL, not panics
- Section headers in both human and JSON output work correctly
- Production checks properly gated behind `opts.Target != "" && s.remoteExecutor != nil`
- All 11 existing tests continue to pass
- 19 new tests cover local runtime and production check scenarios well
- No new dependencies added
- CLI flags (`--target`, `--ssh-key`) properly integrated

**Review by tibtof — COMMENTED** [review]
LGTM (self-authored PR -- cannot formally approve via API). Second review pass -- all 4 previously requested changes have been correctly addressed:

1. **context.Context added to I/O functions**: `checkDomainDNS` and `checkRemoteTLSCert` now accept `ctx context.Context` as their first parameter. DNS resolution uses `net.DefaultResolver.LookupHost(ctx, ...)` and TLS inspection uses `tls.Dialer{}.DialContext(ctx, ...)`, both properly respecting cancellation via derived timeout contexts.

2. **Tests for extractHostFromTarget**: New `doctor_internal_test.go` with a table-driven `TestExtractHostFromTarget` covering 9 cases: SSH URLs with/without user, with/without port, hostnames, IPv6 with brackets and port, bare hostname, and empty string.

3. **url.Parse replaces manual string splitting**: `extractHostFromTarget` now uses `url.Parse` + `u.Hostname()` from `net/url`, correctly handling IPv6 addresses and edge cases.

4. **fakeRemoteExecutor.DryRunTransfer returns empty slice**: Returns `[]string{}, nil` instead of `nil, nil`, matching project conventions.

All tests pass (30+ tests in `internal/app/ops`, including 20 new ones) with race detector enabled. No remaining issues found.
